### ISSUE TITLE
pull failing for k3s Mac multipass hyper-V k8s cluster

### ISSUE THREAD

**Issue by HoffiMuc** [ContainerSolutions/trow#76]
Hi,

I setup a local 3+1 nodes k8s cluster with https://multipass.run and https://k3s.io

source here: https://github.com/HoffiMuc/k3s

I am on a MacBook Laptop and used the quickinstall/install.sh

Everything worked out fine, including `docker push trow.kube-public:31000/minimal_rest:0.1.0`

problems come hereafter:

a) if I do try to pull the image I just pushed again from my local MacBook:

```
$ docker pull trow.kube-public:31000/minimal-rest:0.1.0
Error response from daemon: manifest for trow.kube-public:31000/minimal-rest:0.1.0 not found: manifest unknown: manifest unknown
```

b) pods of my deployment not starting, because:

```
$ k describe pod minimal-rest-deployment-68874c8766-24dpz
... <snip>
Events:
  Type     Reason     Age        From               Message
  ----     ------     ----       ----               -------
  Normal   Scheduled  <unknown>  default-scheduler  Successfully assigned default/minimal-rest-deployment-68874c8766-684jf to node1
  Normal   Pulling    9s         kubelet, node1     Pulling image "trow.kube-public:31000/minimal-rest:0.1.0"
  Warning  Failed     8s         kubelet, node1     Failed to pull image "trow.kube-public:31000/minimal-rest:0.1.0": rpc error: code = Unknown desc = failed to pull and unpack image "trow.kube-public:31000/minimal-rest:0.1.0": failed to resolve reference "trow.kube-public:31000/minimal-rest:0.1.0": failed to do request: Head https://trow.kube-public:31000/v2/minimal-rest/manifests/0.1.0: x509: certificate signed by unknown authority
  Warning  Failed     8s         kubelet, node1     Error: ErrImagePull
  Normal   BackOff    7s         kubelet, node1     Back-off pulling image "trow.kube-public:31000/minimal-rest:0.1.0"
  Warning  Failed     7s         kubelet, node1     Error: ImagePullBackOff
```

any ideas?

**Comment by amouat** [comment]
Did you use the quick install method or the kustomize standard install?

I think the problem is that k3s uses containerd, which has a different way of handling certs than Docker. It's something I need to support but haven't got to yet.

**Comment by HoffiMuc** [comment]
fyi, my k3s installation from last week has `containerd://1.3.3-k3s1`

**Comment by amouat** [comment]
Thanks - that's actually pretty useful. It seems k3s is using the latest version it can get. Which is nice, as the way certs worked with containerd changed recently.

**Comment by amouat** [comment]
So I should be able to fix this when containerd 1.4 is released. I'm going to close this issue and track progress in #78.

Thanks for submitting the issue. In the meantime I'm afraid you'll need to use the standard install which will require using a domain & certs. The only other choices (with any registry) involves editing the containerd config on each node and restarting, or adding certificates to the system store.

Apologies that I don't have a better solution currently. Thanks for submitting the issue.
